### PR TITLE
docs: Update and describe ENV variables

### PR DESCRIPTION
**Pull request by frondeus**
I completely skipped describing leader elector because it's deprecated in favor of distributed DB.

### CODE REVIEW

File: docs/architecture/api.md [hunk 1]
```diff
@@ -25,3 +25,41 @@ Because our schema-registry in docker-compose is automatically initialized with
...
+
+#### Kafka configuration
+(if `COMMUNICATION_METHOD` equals `kafka`)
+
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| KAFKA_BROKERS | Address to kafka brokers | `kafka:9093` | yes | |
+| KAFKA_GROUP_ID | Group id of the consumer | `postgres_command` | yes | |
+| REPORT_SOURCE | kafka topic on which API listens for notifications | `cdl.notifications` | yes | |
+| INSERT_DESTINATION | kafka topic to which API inserts new objects | `cdl.data.input` | yes | |
```
Comment (smores56) [inline docs/architecture/api.md:48]:
```suggestion
| KAFKA_BROKERS | Address to Kafka brokers | `kafka:9093` | yes | |
| KAFKA_GROUP_ID | Group ID of the consumer | `postgres_command` | yes | |
| REPORT_SOURCE | Kafka topic on which API listens for notifications | `cdl.notifications` | yes | |
| INSERT_DESTINATION | Kafka topic to which API inserts new objects | `cdl.data.input` | yes | |
```

File: docs/architecture/api.md [hunk 2]
```diff
@@ -25,3 +25,41 @@ Because our schema-registry in docker-compose is automatically initialized with
...
+
+#### Amqp configuration 
+(if `COMMUNICATION_METHOD` equals `amqp`)
+
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| AMQP_CONNECTION_STRING | Connection string to AMQP Server | `amqp://user:CHANGEME@rabbitmq:5672/%2f` | yes | |
+| AMQP_CONSUMER_TAG | Consumer tag | `postgres_command` | yes | |
+| REPORT_SOURCE | amqp queue on which API listens for notifications | `cdl.notifications` | yes | |
+| INSERT_DESTINATION | amqp exchange to which API inserts new objects | `cdl.data.input` | yes | |
```
Comment (smores56) [inline docs/architecture/api.md:58]:
```suggestion
| AMQP_CONNECTION_STRING | Connection URL to AMQP Server | `amqp://user:CHANGEME@rabbitmq:5672/%2f` | yes | |
| AMQP_CONSUMER_TAG | Consumer tag | `postgres_command` | yes | |
| REPORT_SOURCE | AMQP queue on which API listens for notifications | `cdl.notifications` | yes | |
| INSERT_DESTINATION | AMQP exchange to which API inserts new objects | `cdl.data.input` | yes | |
```

File: docs/architecture/api.md [hunk 3]
```diff
@@ -25,3 +25,41 @@ Because our schema-registry in docker-compose is automatically initialized with
...
+| SCHEMA_REGISTRY_ADDR | Address of setup schema registry | http://schema_registry:50101 | yes | |
+| QUERY_ROUTER_ADDR | Address of query router | http://query_router:50101 | yes | |
+| COMMUNICATION_METHOD | The method to communicate with external services | `kafka` / `amqp` / `grpc` | yes | |
+| RUST_LOG | Log level | `trace` | no | |
+
+#### Kafka configuration
+(if `COMMUNICATION_METHOD` equals `kafka`)
+
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
```
Comment (smores56) [inline docs/architecture/api.md:44]: Not necessary, but I would make at least these dashes the same width of the headers.

File: docs/architecture/api.md [hunk 4]
```diff
@@ -25,3 +25,41 @@ Because our schema-registry in docker-compose is automatically initialized with
...
+| AMQP_CONSUMER_TAG | Consumer tag | `postgres_command` | yes | |
+| REPORT_SOURCE | amqp queue on which API listens for notifications | `cdl.notifications` | yes | |
+| INSERT_DESTINATION | amqp exchange to which API inserts new objects | `cdl.data.input` | yes | |
+
+#### GRPC configuration 
+(if `COMMUNICATION_METHOD` equals `grpc`)
+
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| INSERT_DESTINATION | GRPC Service address on which API inserts new objects | `http://data_router:50101` | yes | |
```
Comment (smores56) [inline docs/architecture/api.md:65]:
```suggestion
| INSERT_DESTINATION | gRPC service address on which API inserts new objects | `http://data_router:50101` | yes | |
```

File: docs/architecture/command_service.md [hunk 5]
```diff
@@ -26,21 +26,65 @@ Egest methods (supported repositories):
 
 ### Configuration (environment files)
```
Comment (smores56) [inline docs/architecture/command_service.md:27]:
```suggestion
### Configuration (Environment Variables)
```

File: docs/architecture/command_service.md [hunk 6]
```diff
@@ -26,21 +26,65 @@ Egest methods (supported repositories):
...
-DRUID_OUTPUT_TOPIC
-REPORT_BROKER
-REPORT_TOPIC
-```
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| COMMUNICATION_METHOD | The method to communicate with external services | `kafka` / `amqp` / `grpc` | yes | |
+| REPORT_DESTINATION | When missing reporting service is disabled. Kafka topic / amqp exchange or REST parameter | `cdl.notifications` | no | |
+| METRICS_PORT | Port to listen on prometheus requests | 58105 | no | 58105 |
+| RUST_LOG | Log level | `trace` | no | |
```
Comment (smores56) [inline docs/architecture/command_service.md:34]:
```suggestion
| COMMUNICATION_METHOD | The method to communicate with external services | `kafka` / `amqp` / `grpc` | yes | |
| REPORT_DESTINATION | Kafka topic/AMQP exchange/callback URL to send notifications to (reporting disabled when empty) | `cdl.notifications` | no | |
| METRICS_PORT | Port to listen on for Prometheus requests | 58105 | no | 58105 |
| RUST_LOG | Log level | `trace` | no | |
```

File: docs/architecture/command_service.md [hunk 7]
```diff
@@ -26,21 +26,65 @@ Egest methods (supported repositories):
...
+
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| DRUID_OUTPUT_BROKERS | Kafka brokers | `kafka:9093` | yes | |
+| DRUID_OUTPUT_TOPIC | Kafka topic | `cdl.timeseries.internal.druid` | yes | |
+
+#### Victoria metrics configuration
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| VICTORIA_METRICS_OUTPUT_URL | Url to the VM | `http://victoria_metrics:8428` | yes | |
```
Comment (smores56) [inline docs/architecture/command_service.md:57]:
```suggestion
| VICTORIA_METRICS_OUTPUT_URL | URL to the VM | `http://victoria_metrics:8428` | yes | |
```

File: docs/architecture/command_service.md [hunk 8]
```diff
@@ -26,21 +26,65 @@ Egest methods (supported repositories):
...
+#### Kafka configuration 
+(if `COMMUNICATION_METHOD` equals `kafka`)
+
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| KAFKA_BROKERS | Address to kafka brokers | `kafka:9093` | yes | |
+| KAFKA_GROUP_ID | Group id of the consumer | `postgres_command` | yes | |
+| ORDERED_SOURCES | topics with ordered messages | `cdl.timeseries.vm.1.data` | no, but one of `ORDERED_SOURCES` and `UNORDERED_SOURCES` has to be present | |
+| UNORDERED_SOURCES | topics with unordered messages | `cdl.timeseries.vm.2.data` | no, but one of `ORDERED_SOURCES` and `UNORDERED_SOURCES` has to be present | |
+| TASK_LIMIT | limits the number of tasks | 32 | yes | 32 |
```
Comment (smores56) [inline docs/architecture/command_service.md:68]:
```suggestion
| KAFKA_BROKERS | Address of Kafka brokers | `kafka:9093` | yes | |
| KAFKA_GROUP_ID | Group ID of the consumer | `postgres_command` | yes | |
| ORDERED_SOURCES | Topics with ordered messages | `cdl.timeseries.vm.1.data` | no, but one of `ORDERED_SOURCES` and `UNORDERED_SOURCES` has to be present | |
| UNORDERED_SOURCES | Topics with unordered messages | `cdl.timeseries.vm.2.data` | no, but one of `ORDERED_SOURCES` and `UNORDERED_SOURCES` has to be present | |
| TASK_LIMIT | Max requests handled in parallel | 32 | yes | 32 |
```

File: docs/architecture/command_service.md [hunk 9]
```diff
@@ -26,21 +26,65 @@ Egest methods (supported repositories):
...
+| DRUID_OUTPUT_BROKERS | Kafka brokers | `kafka:9093` | yes | |
+| DRUID_OUTPUT_TOPIC | Kafka topic | `cdl.timeseries.internal.druid` | yes | |
+
+#### Victoria metrics configuration
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| VICTORIA_METRICS_OUTPUT_URL | Url to the VM | `http://victoria_metrics:8428` | yes | |
+
+#### Kafka configuration 
+(if `COMMUNICATION_METHOD` equals `kafka`)
```
Comment (smores56) [inline docs/architecture/command_service.md:60]: This should be in _italix_ (IMHO), as should all others in the book.

File: docs/architecture/command_service.md [hunk 10]
```diff
@@ -26,21 +26,65 @@ Egest methods (supported repositories):
...
+#### Amqp configuration 
+(if `COMMUNICATION_METHOD` equals `amqp`)
+
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| AMQP_CONNECTION_STRING | Connection string to AMQP Server | `amqp://user:CHANGEME@rabbitmq:5672/%2f` | yes | |
+| AMQP_CONSUMER_TAG | Consumer tag | `postgres_command` | yes | |
+| ORDERED_SOURCES | queus with ordered messages | `cdl.timeseries.vm.1.data` | no, but one of `ORDERED_SOURCES` and `UNORDERED_SOURCES` has to be present | |
+| UNORDERED_SOURCES | queues with unordered messages | `cdl.timeseries.vm.2.data` | no, but one of `ORDERED_SOURCES` and `UNORDERED_SOURCES` has to be present | |
+| TASK_LIMIT | limits the number of tasks | 32 | yes | 32 |
```
Comment (smores56) [inline docs/architecture/command_service.md:79]:
```suggestion
| AMQP_CONNECTION_STRING | Connection URL to AMQP Server | `amqp://user:CHANGEME@rabbitmq:5672/%2f` | yes | |
| AMQP_CONSUMER_TAG | Consumer tag | `postgres_command` | yes | |
| ORDERED_SOURCES | Queues with ordered messages | `cdl.timeseries.vm.1.data` | no, but one of `ORDERED_SOURCES` and `UNORDERED_SOURCES` has to be present | |
| UNORDERED_SOURCES | Queues with unordered messages | `cdl.timeseries.vm.2.data` | no, but one of `ORDERED_SOURCES` and `UNORDERED_SOURCES` has to be present | |
| TASK_LIMIT | Max requests handled in parallel | 32 | yes | 32 |
```

File: docs/architecture/command_service.md [hunk 11]
```diff
@@ -26,21 +26,65 @@ Egest methods (supported repositories):
...
+
+#### GRPC configuration 
+(if `COMMUNICATION_METHOD` equals `grpc`)
+
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+|
+| GRPC_PORT | Port to listen on | 50103 | yes | |
+| REPORT_ENDPOINT_URL | Url to send notifications on | `notifications:50102` | yes | |
```
Comment (smores56) [inline docs/architecture/command_service.md:88]:
```suggestion
| GRPC_PORT | Port to listen on | 50103 | yes | |
| REPORT_ENDPOINT_URL | URL to send notifications to | `notifications:50102` | yes | |
```

File: docs/architecture/data_router.md [hunk 12]
```diff
@@ -62,14 +62,39 @@ Please mind that internally, each message will get its own timestamp, with which
 To configure data router, set following environment variables. `INPUT_ADDR` and `INPUT_QUEUE` is related to the incoming data in the router. `KAFKA_BROKERS`, `KAFKA_ERROR_CHANNEL` are related to messages being routed through Kafka to the corresponding command service.
```
Comment (smores56) [inline docs/architecture/data_router.md:62]:
```suggestion
To configure the Data Router, set the following environment variables:
```

File: docs/architecture/data_router.md [hunk 13]
```diff
@@ -62,14 +62,39 @@ Please mind that internally, each message will get its own timestamp, with which
...
-```
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| COMMUNICATION_METHOD | The method to communicate with external services | `kafka` / `amqp` / `grpc` | yes | |
+| INPUT_SOURCE | kafka topic or amqp queue | `cdl.data.input` | no, when `grpc` has been chosen | |
+| SCHEMA_REGISTRY_ADDR | Address of setup schema registry | http://schema_registry:50101 | yes | |
+| CACHE_CAPACITY | How many entries the cache can hold | 1024 | yes | |
+| TASK_LIMIT | limits the number of tasks | 128 | yes | 128 |
+| METRICS_PORT | Port to listen on prometheus requests | 58105 | no | 58105 |
+| RUST_LOG | Log level | `trace` | no | |
```
Comment (smores56) [inline docs/architecture/data_router.md:73]:
```suggestion
| COMMUNICATION_METHOD | The method to communicate with external services | `kafka` / `amqp` / `grpc` | yes | |
| INPUT_SOURCE | Kafka topic or AMQP queue | `cdl.data.input` | no, when `grpc` has been chosen | |
| SCHEMA_REGISTRY_ADDR | Address of schema registry gRPC API | http://schema_registry:50101 | yes | |
| CACHE_CAPACITY | How many entries the cache can hold | 1024 | yes | |
| TASK_LIMIT | Max requests handled in parallel | 128 | yes | 128 |
| METRICS_PORT | Port to send metrics data to Prometheus over | 58105 | no | 58105 |
| RUST_LOG | Log level | `trace` | no | |
```

File: docs/architecture/data_router.md [hunk 14]
```diff
@@ -62,14 +62,39 @@ Please mind that internally, each message will get its own timestamp, with which
...
+
+#### Kafka configuration 
+(if `COMMUNICATION_METHOD` equals `kafka`)
+
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| KAFKA_BROKERS | Address to kafka brokers | `kafka:9093` | yes | |
+| KAFKA_GROUP_ID | Group id of the consumer | `data_router` | yes | |
```
Comment (smores56) [inline docs/architecture/data_router.md:81]:
```suggestion
| KAFKA_BROKERS | Address of Kafka brokers | `kafka:9093` | yes | |
| KAFKA_GROUP_ID | Group ID of the consumer | `data_router` | yes | |
```

File: docs/architecture/data_router.md [hunk 15]
```diff
@@ -62,14 +62,39 @@ Please mind that internally, each message will get its own timestamp, with which
...
+
+#### Amqp configuration 
+(if `COMMUNICATION_METHOD` equals `amqp`)
+
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| AMQP_CONNECTION_STRING | Connection string to AMQP Server | `amqp://user:CHANGEME@rabbitmq:5672/%2f` | yes | |
```
Comment (smores56) [inline docs/architecture/data_router.md:88]:
```suggestion
| AMQP_CONNECTION_STRING | Connection URL to AMQP Server | `amqp://user:CHANGEME@rabbitmq:5672/%2f` | yes | |
```

File: docs/architecture/data_router.md [hunk 16]
```diff
@@ -62,14 +62,39 @@ Please mind that internally, each message will get its own timestamp, with which
...
+| AMQP_CONSUMER_TAG | Consumer tag | `data_router` | yes | |
+
+#### GRPC configuration 
+(if `COMMUNICATION_METHOD` equals `grpc`)
+
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+|
```
Comment (smores56) [inline docs/architecture/data_router.md:96]: Extra line here.

File: docs/architecture/query_router.md [hunk 17]
```diff
@@ -21,24 +21,17 @@ Communication protocols:
...
-CACHE_CAPACITY
-RUST_LOG
-`
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| INPUT_PORT | Port to listen on | 50103 | yes | |
+| SCHEMA_REGISTRY_ADDR | Address of setup schema registry | http://schema_registry:50101 | yes | |
+| CACHE_CAPACITY | How many entries the cache can hold | 1024 | yes | |
+| METRICS_PORT | Port to listen on prometheus requests | 58105 | no | 58105 |
+| RUST_LOG | Log level | `trace` | no | |
```
Comment (smores56) [inline docs/architecture/query_router.md:30]:
```suggestion
| INPUT_PORT | Port to listen on | 50103 | yes | |
| SCHEMA_REGISTRY_ADDR | Address of schema registry gRPC API | http://schema_registry:50101 | yes | |
| CACHE_CAPACITY | How many entries the cache can hold | 1024 | yes | |
| METRICS_PORT | Port to send metrics data to Prometheus over | 58105 | no | 58105 |
| RUST_LOG | Log level | `trace` | no | |
```

File: docs/architecture/query_service.md [hunk 18]
```diff
@@ -26,16 +26,22 @@ Communication protocols:
...
-POSTGRES_HOST
-POSTGRES_PORT
-POSTGRES_DBNAME
-POSTGRES_SCHEMA
-`
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| INPUT_PORT | Port to listen on | 50103 | yes | |
+| METRICS_PORT | Port to listen on prometheus requests | 58105 | no | 58105 |
+| RUST_LOG | Log level | `trace` | no | |
```
Comment (smores56) [inline docs/architecture/query_service.md:33]:
```suggestion
| INPUT_PORT | Port to listen on | 50103 | yes | |
| METRICS_PORT | Port to send metrics data to Prometheus over | 58105 | no | 58105 |
| RUST_LOG | Log level | `trace` | no | |
```

File: docs/architecture/schema_registry.md [hunk 19]
```diff
@@ -22,16 +22,41 @@ Communication methods (supported repositores):
...
+|---|---|---|---|---|
+| KAFKA_BROKERS | Address to kafka brokers | `kafka:9093` | yes | |
+| KAFKA_GROUP_ID | Group id of the consumer | `schema_registry` | yes | |
+
+#### Amqp configuration 
+(if `COMMUNICATION_METHOD` equals `amqp`)
+
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| AMQP_CONNECTION_STRING | Connection string to AMQP Server | `amqp://user:CHANGEME@rabbitmq:5672/%2f` | yes | |
```
Comment (smores56) [inline docs/architecture/schema_registry.md:50]:
```suggestion
| AMQP_CONNECTION_STRING | Connection URL to AMQP Server | `amqp://user:CHANGEME@rabbitmq:5672/%2f` | yes | |
```

File: docs/architecture/schema_registry.md [hunk 20]
```diff
@@ -22,16 +22,41 @@ Communication methods (supported repositores):
...
+| METRICS_PORT | Port to listen on prometheus requests | 58105 | no | 58105 |
+| RUST_LOG | Log level | `trace` | no | |
+
+#### Kafka configuration 
+(if `COMMUNICATION_METHOD` equals `kafka`)
+
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| KAFKA_BROKERS | Address to kafka brokers | `kafka:9093` | yes | |
+| KAFKA_GROUP_ID | Group id of the consumer | `schema_registry` | yes | |
```
Comment (smores56) [inline docs/architecture/schema_registry.md:43]:
```suggestion
| KAFKA_BROKERS | Address to Kafka brokers | `kafka:9093` | yes | |
| KAFKA_GROUP_ID | Group ID of the consumer | `schema_registry` | yes | |
```

File: docs/architecture/schema_registry.md [hunk 21]
```diff
@@ -22,16 +22,41 @@ Communication methods (supported repositores):
...
+| AMQP_CONNECTION_STRING | Connection string to AMQP Server | `amqp://user:CHANGEME@rabbitmq:5672/%2f` | yes | |
+| AMQP_CONSUMER_TAG | Consumer tag | `schema_registry` | yes | |
+
+#### Replication configuration 
+(if `COMMUNICATION_METHOD` does **not** equal `grpc`)
+
+| Name | Short Description | Example | Mandatory | Default |
+|---|---|---|---|---|
+| REPLICATION_SOURCE | kafka topic/amqp queue | `cdl.schema_registry.internal` | yes | |
+| REPLICATION_DESTINATION | kafka topic/amqp exchange | `cdl.schema_registry.internal` | yes | |
```
Comment (smores56) [inline docs/architecture/schema_registry.md:59]:
```suggestion
| REPLICATION_SOURCE | Kafka topic/AMQP queue | `cdl.schema_registry.internal` | yes | |
| REPLICATION_DESTINATION | Kafka topic/AMQP exchange | `cdl.schema_registry.internal` | yes | |
```

File: docs/architecture/schema_registry.md [hunk 22]
```diff
@@ -22,16 +22,41 @@ Communication methods (supported repositores):
...
+|---|---|---|---|---|
+| INPUT_PORT | Port to listen on | 50103 | yes | |
+| COMMUNICATION_METHOD | The method to communicate with external services | `kafka` / `amqp` / `grpc` | yes | |
+| REPLICATION_ROLE | (deprecated) | `master` / `slave` / `none` | yes | |
+| DB_NAME | Database name | `schema-registry` | yes | |
+| POD_NAME | (deprecated) used to promote to `master` | `schema1` | no | |
+| EXPORT_DIR | Directory to save state of the database. The state is saved in newly created folder with timestamp | `/var/db` | no | |
+| IMPORT_FILE | JSON file from which SR should load initial state. If the state already exists this env variable will be ignored | `/var/db/initial-schema.json` | no | |
+| METRICS_PORT | Port to listen on prometheus requests | 58105 | no | 58105 |
+| RUST_LOG | Log level | `trace` | no | |
```
Comment (smores56) [inline docs/architecture/schema_registry.md:35]:
```suggestion
| INPUT_PORT | Port to listen on | 50103 | yes | |
| COMMUNICATION_METHOD | The method to communicate with external services | `kafka` / `amqp` / `grpc` | yes | |
| REPLICATION_ROLE | _(deprecated)_ | `master` / `slave` / `none` | yes | |
| DB_NAME | Database name | `schema-registry` | yes | |
| POD_NAME | _(deprecated)_ used to promote to `master` | `schema1` | no | |
| EXPORT_DIR | Directory to save state of the database. The state is saved in newly created folder with timestamp | `/var/db` | no | |
| IMPORT_FILE | JSON file from which SR should load initial state. If the state already exists this env variable will be ignored | `/var/db/initial-schema.json` | no | |
| METRICS_PORT | Port to send metrics data to Prometheus over | 58105 | no | 58105 |
| RUST_LOG | Log level | `trace` | no | |
```

**Review by smores56 — DISMISSED** [review]

**Review by esavier — DISMISSED** [review]

**Review by esavier — APPROVED** [review]

**Pull request merged**